### PR TITLE
CoAP option update

### PR DESCRIPTION
**Pull request by GiedriusM**
* Add GetFirstOption
* Remove GetCurrentOption
* Change corresponding APIs
* Do not drop requests that have unknown CoAP options

### CODE REVIEW

**Comment by GiedriusM** [comment]
This also solves #1522

File: src/core/coap/coap_server.cpp [hunk 1]
```diff
@@ -173,7 +173,7 @@ void Server::ProcessReceivedMessage(Message &aMessage, const Ip6::MessageInfo &a
         break;
     }
 
-    coapOption = header.GetCurrentOption();
+    coapOption = header.GetFirstOption();
```
Comment (GiedriusM) [inline src/core/coap/coap_server.cpp]: This also fixes a possible bug:
When payload with no option arrives, GetCurrentOption would return non-null value, i.e. pointer to mOption, which, in this case, would be uninitialized and could result in unexpected behavior.
Maybe Init() should be called in FromMessage?

Comment (jwhui) [inline src/core/coap/coap_server.cpp]: I agree it would be safer to call `Init()` in `FromMessage()`.

File: include/openthread/coap.h [hunk 2]
```diff
@@ -344,14 +345,14 @@ uint8_t otCoapHeaderGetTokenLength(const otCoapHeader *aHeader);
 const uint8_t *otCoapHeaderGetToken(const otCoapHeader *aHeader);
 
 /**
- * This function returns a pointer to the current option.
+ * This function returns a pointer to the first option.
  *
  * @param[in]  aHeader  A pointer to the CoAP header.
  *
- * @returns A pointer to the current option. If no option is present NULL pointer is returned.
+ * @returns A pointer to the first option. If no more options are present NULL pointer is returned.
```
Comment (jwhui) [inline include/openthread/coap.h]: Should `If no more options are present` be `If no options are present`?

Comment (GiedriusM) [inline include/openthread/coap.h]: Agreed

**Pull request merged**